src/sentry/static/sentry/app/views/settings/components/dataScrubbing/modals/modalManager.tsx

```diff
@@ -0,0 +1,301 @@
+import React from 'react';
+import omit from 'lodash/omit';
+import isEqual from 'lodash/isEqual';
+
+import {addErrorMessage} from 'app/actionCreators/indicator';
+import {Client} from 'app/api';
+import {t} from 'app/locale';
+import {ModalRenderProps} from 'app/actionCreators/modal';
+import {Organization, Project} from 'app/types';
+
+import {
+  RuleType,
+  MethodType,
+  Rule,
+  ProjectId,
+  KeysOfUnion,
+  EventIdStatus,
+} from '../types';
+import submitRules from '../submitRules';
+import Form from './form';
+import Modal from './modal';
+import handleError from './handleError';
+import {valueSuggestions} from '../utils';
+import {fetchSourceGroupData, saveToSourceGroupData} from './utils';
+
+type FormProps = React.ComponentProps<typeof Form>;
+type Values = FormProps['values'];
+type EventId = NonNullable<FormProps['eventId']>;
+type SourceSuggestions = FormProps['sourceSuggestions'];
+
+type Props<T extends ProjectId> = ModalRenderProps & {
+  onSubmitSuccess: (data: T extends undefined ? Organization : Project) => void;
+  orgSlug: Organization['slug'];
+  api: Client;
+  endpoint: string;
+  savedRules: Array<Rule>;
+  projectId?: T;
+};
+
+type State = {
+  values: Values;
+  requiredValues: Array<keyof Values>;
+  errors: FormProps['errors'];
+  isFormValid: boolean;
+  title: string;
+  sourceSuggestions: SourceSuggestions;
+  eventId: EventId;
+};
+
+class ModalManager<
+  T extends ProjectId = undefined,
+  P extends Props<T> = Props<T>,
+  S extends State = State
+> extends React.Component<P, S> {
+  constructor(props: P) {
+    super(props);
+    this.handleSave = this.handleSave.bind(this);
```

Comment:
This is not needed since it's already bound when declaring the method

Reply:
sure.. but I was going to remove the arrow functions since they are much slower than bound functions  when used in a class

Reply:
How are they much slower?

Reply:
![image](https://user-images.githubusercontent.com/1421724/86046372-f49ec180-ba01-11ea-8ff2-fc24afae7c4c.png)

It's true they are slower.. but IMO that performance difference is going to be more than negligible.

Use them when you need to pass around an instance method (like to onClick), but don't use them if you're only calling your method from the class. That's always been my rule. Not sure if we can enforce that with eslint.